esphome/components/light/__init__.py

```diff
@@ -145,6 +145,28 @@
             [cv.percentage], cv.Length(min=3, max=4)
         ),
         cv.Optional(CONF_POWER_SUPPLY): cv.use_id(power_supply.PowerSupply),
+        cv.Optional("power_limits"): cv.Schema(
+            {
+                cv.Required("max_watts"): cv.positive_float,
+                cv.Optional("voltage", default=5.0): cv.positive_float,
```

Comment:
```suggestion
                cv.Optional("voltage", default=5.0): cv.voltage,
```
And maybe add a validator for Watts, or use Ampere instead for the other configuration values?